app/commands/exercise/representation/update_num_submissions.rb

```diff
@@ -0,0 +1,29 @@
+class Exercise::Representation::UpdateNumSubmissions
+  include Mandate
+
+  queue_as :default
+
+  initialize_with :representation
+
+  def call
+    # We're updating in a single query instead of two queries to avoid race-conditions
+    # and using read_committed to avoid deadlocks
+    ActiveRecord::Base.transaction(isolation: Exercism::READ_COMMITTED) do
+      Exercise::Representation.
+        where(id: representation.id).
+        update_all("num_submissions = (#{num_submissions_sql})")
```

Comment:
I have a **slight** bias here to this being `num_iterations`. As we only create representations for the submissions of representations, I feel like it's a clearer name as to the intention of the column (even though it's measuring submissions under the hood). But my bias is only very slight, so feel free to overrule it if you prefer `num_submissions`.

Reply:
> As we only create representations for the submissions of representations

Did you mean to write:

> As we only create representations for the iterations of submissions

?

Reply:
I mean to write "for the submissions of iterations". Sorry.